src/sentry/static/sentry/app/views/eventsV2/querycard.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import styled from 'react-emotion';
+import space from 'app/styles/space';
+import Link from 'app/components/links/link';
+import InlineSvg from 'app/components/inlineSvg';
+
+type Props = {
+  title?: string;
+  queryDetail?: string;
+  to?: string | object;
+  onEventClick?: () => void;
+};
+
+class QueryCard extends React.Component<Props> {
+  render() {
+    const {title, queryDetail, onEventClick, to} = this.props;
+
+    return (
+      <StyledQueryCard onClick={onEventClick} to={to}>
+        <QueryCardHeader>
+          <StyledTitle>{title}</StyledTitle>
+          <StyledQueryDetail>{queryDetail}</StyledQueryDetail>
+        </QueryCardHeader>
+        <QueryCardBody />
+        <QueryCardFooter>
+          <StyledCreator>
+            <small>Pre-Built Query</small>
+          </StyledCreator>
+          <InlineSvg src="icon-ellipsis-filled" />
+        </QueryCardFooter>
+      </StyledQueryCard>
+    );
+  }
+}
```

Comment:
If `state` will not be used, it might be better to be use `React.PureComponent` or change this to a function component instead.

Reply:
This doesn't matter at this point but if there are a lot of `QueryCard` being rendered, using PureComponent or function component will improve  performance.

Reply:
I'll go ahead and change it to PureComponent